### PR TITLE
External access to hermes

### PR DESCRIPTION
**Pull request by jeanleonov**
- Added ability to run multiple services on a single nginx port but different locations.
 - Using this feature, admin server shares nginx port 17441 with hermes (only /stats/cluster methods are available externally).

https://ocd.appscale.com:8080/job/Daily%20Build/3528/
https://ocd.appscale.com:8080/job/Hawkeye-Simple-8/578/ (one more try for hawkeye test).

### CODE REVIEW

**Comment by jeanleonov** [comment]
https://ocd.appscale.com:8080/job/Daily%20Build/3535/
After resolving codacy issues.

**Comment by jeanleonov** [comment]
https://ocd.appscale.com:8080/job/Daily%20Build/3535/ (tools failed)
https://ocd.appscale.com:8080/job/Tools-test/2127/ (one more try)

File: AppController/lib/nginx.rb [hunk 1]
```diff
@@ -341,16 +342,41 @@ def self.clear_sites_enabled()
...
     nginx_port, location='/')
+    proxy_pass = "#{service_host}:#{service_port}"
+    config_path = File.join(SITES_ENABLED_PATH,
+                            "#{service_name}.#{CONFIG_EXTENSION}")
+    old_config = File.read(config_path) if File.file?(config_path)
+
+    locations = {}
+
+    if old_config
+      # Check
```
Comment (cdonati) [inline AppController/lib/nginx.rb]: cryptic comment

File: AppController/djinn.rb [hunk 2]
```diff
@@ -3746,6 +3746,13 @@ def start_hermes()
       @options["write_detailed_processes_stats_log"].downcase == 'true',
       @options["write_detailed_proxies_stats_log"].downcase == 'true'
     )
+    if my_node.is_shadow?
+      nginx_port = 17441
+      service_port = 4378
+      Nginx.add_service_location(
+        'appscale-administration', my_node.private_ip,
+        service_port, nginx_port, '/stats/cluster/')
```
Comment (whoarethebritons) [inline AppController/djinn.rb]: I'm a bit confused, why are we adding service location `appscale-administration` twice to shadow with different paths

Comment (jeanleonov) [inline AppController/djinn.rb]: `"appscale-administration"`is a service name. This service has two locations:
`"/stats/cluster"` which is handled by Hermes and `"/"` (other URL paths) which is handled by Admin Server.
So locations are different, but they works over the same port and they are parts of the same service.

**Comment by jeanleonov** [comment]
https://ocd.appscale.com:8080/job/Daily%20Build/3587/
2 failed hawkeye tests seems to be irrelevant.

**Pull request merged**